ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -27,8 +27,261 @@
+        upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 1);
+        objects.clear();
+
+        // Delete the object we uploaded and test list_objects, should return an empty objects list.
+        client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.empty());
+        objects.clear();
+
+        // Upload multiple files and test list.
+        const int32_t total_objects = 20;
+        for (int i = 0; i < total_objects; i++) {
+            std::string multi_file_name = object_name + std::to_string(i);
+            REQUIRE(
+              upload_file(client, test_bucket_name, test_bucket_prefix, file_name, multi_file_name).ok());
+        }
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == total_objects);
+        objects.clear();
+
+        // Test if list single only returns one object.
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 1);
+        objects.clear();
+
+        // Delete all object we uploaded.
+        for (int i = 0; i < total_objects; i++) {
+            client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name + std::to_string(i));
+        }
+
+        // Bucket should be cleared.
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.empty());
+    }
+
+    SECTION("Simple put test", "[gcp-connection]")
+    {
+
+        // Upload a file that does not exist locally - should fail.
+        REQUIRE(conn.put_object(non_existant_object_name, non_existant_file_name) == ENOENT);
+
+        // Check number of files with the given prefix that are currently in the bucket.
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 0));
+
+        // Upload a test file.
+        REQUIRE(conn.put_object(object_name, "./" + file_name) == 0);
+
+        REQUIRE(check_object_num(client, test_bucket_name, test_bucket_prefix, 1));
+
+        // Check the bucket contains the uploaded file.
+        bool exists = check_exist(client, test_bucket_name, test_bucket_prefix, object_name);
+        REQUIRE(exists);
+
+        // Delete the uploaded file.
+        client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+
+        exists = check_exist(client, test_bucket_name, test_bucket_prefix, object_name);
+        REQUIRE_FALSE(exists);
+    }
+
+    SECTION("Simple delete test", "[gcp-connection]")
+    {
+
+        // Delete a file that does not exist in the bucket - should fail.
+        REQUIRE(conn.delete_object(non_existant_object_name) == ENOENT);
+
+        // Upload a test file.
+        auto metadata = upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name);
```

Comment:
This can just be `REQUIRE(upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name).ok())`

Reply:
addressed here: [8039651](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/8039651071462d3bd21d784fdb335f39875e742f)